lib/index.js

```diff
@@ -159,28 +151,63 @@ function combineMediaQuery(queries, resolutions) {
   }, []).join(', ');
 }
 
-// Matches `url()` content as long as it is followed by `at-2x`
-const urlPathRegex = /url\(([^\r\n]+)\)(?:[^\r\n]+)?at-2x/gm;
+function createRetinaImages(bgValue, identifier) {
+  const backgrounds = splitMultipleBackgrounds(bgValue);
+  const images = backgrounds.map(extractRetinaImage.bind(this, identifier));
+  return images.join(', ');
+}
+
+// Matches the <image> type and value within a background definition
+const imageRegex = /([^\s]+)\((.+)\)/;
 
-function createRetinaUrl(bgValue, identifier) {
-  let match;
-  // Loop over all occurances of `url()` and match the path
-  while ((match = urlPathRegex.exec(bgValue)) !== null) {
-    const [, imgUrl] = match;
-    const extension = path.extname(imgUrl);
+// Returns the <image> part of the background definition,
+// and includes the identifier if it meets the criteria:
+// * It's a url() image
+// * It's not an svg
+// * The background definition has at-2x applied
+function extractRetinaImage(identifier, background) {
+  const match = background.match(imageRegex);
 
-    if (extension === '.svg') {
-      break;
-    }
+  if (!match) {
+    return background;
+  }
+
+  const [image, type, value] = match;
+
+  if (background.indexOf('at-2x') === -1 || type !== 'url') {
+    return image;
+  }
 
-    // File name without extension
-    const filename = path.basename(path.basename(imgUrl), extension);
+  const extension = path.extname(value);
 
-    // Replace with retina filename
-    bgValue = bgValue.replace(filename + extension, filename + identifier + extension);
+  if (extension === '.svg') {
+    return image;
   }
 
-  return removeKeyword(bgValue);
+  // File name without extension
+  const filename = path.basename(path.basename(value), extension);
+  // Replace with retina filename
+  return image.replace(filename + extension, filename + identifier + extension);
+}
+
+function splitMultipleBackgrounds(value) {
+  const backgrounds = [];
+  let opened = 0;
+  value.split(',').forEach((part) => {
+    if (opened > 0) {
+      backgrounds[backgrounds.length - 1] += `,${part}`;
+    } else {
+      backgrounds.push(part);
+    }
+    const open = count(part, '(');
+    const close = count(part, ')');
+    opened += open - close;
+  });
+  return backgrounds;
+}
```

Comment:
@tdhooper Curious about the intention of this function. Could the same not be achieved with `return value.split(',')`?